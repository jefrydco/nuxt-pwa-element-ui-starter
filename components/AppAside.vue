<template>
  <el-aside width="200px" style="background-color: rgb(238, 241, 246)">
    <el-menu :default-openeds="['1', '3']">
      <el-submenu index="1">
        <template #title="">
          <i class="el-icon-message"></i>Navigator One
        </template>
        <el-menu-item-group>
          <template #title=""> Group 1 </template>
          <el-menu-item index="1-1">Option 1</el-menu-item>
          <el-menu-item index="1-2">Option 2</el-menu-item>
        </el-menu-item-group>
        <el-menu-item-group title="Group 2">
          <el-menu-item index="1-3">Option 3</el-menu-item>
        </el-menu-item-group>
        <el-submenu index="1-4">
          <template #title=""> Option4 </template>
          <el-menu-item index="1-4-1">Option 4-1</el-menu-item>
        </el-submenu>
      </el-submenu>
      <el-submenu index="2">
        <template #title="">
          <i class="el-icon-menu"></i>Navigator Two
        </template>
        <el-menu-item-group>
          <template #title=""> Group 1 </template>
          <el-menu-item index="2-1">Option 1</el-menu-item>
          <el-menu-item index="2-2">Option 2</el-menu-item>
        </el-menu-item-group>
        <el-menu-item-group title="Group 2">
          <el-menu-item index="2-3">Option 3</el-menu-item>
        </el-menu-item-group>
        <el-submenu index="2-4">
          <template #title=""> Option 4 </template>
          <el-menu-item index="2-4-1">Option 4-1</el-menu-item>
        </el-submenu>
      </el-submenu>
      <el-submenu index="3">
        <template #title="">
          <i class="el-icon-setting"></i>Navigator Three
        </template>
        <el-menu-item-group>
          <template #title=""> Group 1 </template>
          <el-menu-item index="3-1">Option 1</el-menu-item>
          <el-menu-item index="3-2">Option 2</el-menu-item>
        </el-menu-item-group>
        <el-menu-item-group title="Group 2">
          <el-menu-item index="3-3">Option 3</el-menu-item>
        </el-menu-item-group>
        <el-submenu index="3-4">
          <template #title=""> Option 4 </template>
          <el-menu-item index="3-4-1">Option 4-1</el-menu-item>
        </el-submenu>
      </el-submenu>
    </el-menu>
  </el-aside>
</template>

<script>
export default {
  data() {
    return {}
  }
}
</script>
