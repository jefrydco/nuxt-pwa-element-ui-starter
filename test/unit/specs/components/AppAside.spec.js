import { shallowMount, createLocalVue } from "@vue/test-utils";
import ElementUI from "element-ui";

import AppAside from "~/components/AppAside";

const localVue = createLocalVue();
localVue.use(ElementUI);

describe("App Aside", () => {
  it("renders to match snapshot", () => {
    const wrapper = shallowMount(AppAside, {
      localVue
    });
    expect(wrapper).toMatchSnapshot();
  });
});
