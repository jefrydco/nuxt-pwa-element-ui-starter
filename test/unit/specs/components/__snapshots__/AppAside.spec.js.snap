// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Aside renders to match snapshot 1`] = `
<el-aside-stub width="200px" style="background-color: rgb(238, 241, 246);">
  <el-menu-stub mode="vertical" defaultactive="" defaultopeneds="1,3" menutrigger="hover" collapsetransition="true">
    <el-submenu-stub offset="0" boundariespadding="5" popperoptions="[object Object]" index="1" showtimeout="300" hidetimeout="300">
      <el-menu-item-group-stub>
        <el-menu-item-stub index="1-1">Option 1</el-menu-item-stub>
        <el-menu-item-stub index="1-2">Option 2</el-menu-item-stub>
      </el-menu-item-group-stub>
      <el-menu-item-group-stub title="Group 2">
        <el-menu-item-stub index="1-3">Option 3</el-menu-item-stub>
      </el-menu-item-group-stub>
      <el-submenu-stub offset="0" boundariespadding="5" popperoptions="[object Object]" index="1-4" showtimeout="300" hidetimeout="300">
        <el-menu-item-stub index="1-4-1">Option 4-1</el-menu-item-stub>
      </el-submenu-stub>
    </el-submenu-stub>
    <el-submenu-stub offset="0" boundariespadding="5" popperoptions="[object Object]" index="2" showtimeout="300" hidetimeout="300">
      <el-menu-item-group-stub>
        <el-menu-item-stub index="2-1">Option 1</el-menu-item-stub>
        <el-menu-item-stub index="2-2">Option 2</el-menu-item-stub>
      </el-menu-item-group-stub>
      <el-menu-item-group-stub title="Group 2">
        <el-menu-item-stub index="2-3">Option 3</el-menu-item-stub>
      </el-menu-item-group-stub>
      <el-submenu-stub offset="0" boundariespadding="5" popperoptions="[object Object]" index="2-4" showtimeout="300" hidetimeout="300">
        <el-menu-item-stub index="2-4-1">Option 4-1</el-menu-item-stub>
      </el-submenu-stub>
    </el-submenu-stub>
    <el-submenu-stub offset="0" boundariespadding="5" popperoptions="[object Object]" index="3" showtimeout="300" hidetimeout="300">
      <el-menu-item-group-stub>
        <el-menu-item-stub index="3-1">Option 1</el-menu-item-stub>
        <el-menu-item-stub index="3-2">Option 2</el-menu-item-stub>
      </el-menu-item-group-stub>
      <el-menu-item-group-stub title="Group 2">
        <el-menu-item-stub index="3-3">Option 3</el-menu-item-stub>
      </el-menu-item-group-stub>
      <el-submenu-stub offset="0" boundariespadding="5" popperoptions="[object Object]" index="3-4" showtimeout="300" hidetimeout="300">
        <el-menu-item-stub index="3-4-1">Option 4-1</el-menu-item-stub>
      </el-submenu-stub>
    </el-submenu-stub>
  </el-menu-stub>
</el-aside-stub>
`;
